flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSplitGenerator.java

```diff
@@ -37,9 +40,20 @@ private FlinkSplitGenerator() {
   static FlinkInputSplit[] createInputSplits(Table table, ScanContext context) {
     List<CombinedScanTask> tasks = tasks(table, context);
     FlinkInputSplit[] splits = new FlinkInputSplit[tasks.size()];
-    for (int i = 0; i < tasks.size(); i++) {
-      splits[i] = new FlinkInputSplit(i, tasks.get(i));
-    }
+    boolean localityPreferred = context.locality();
+
+    Tasks.range(tasks.size())
+        .stopOnFailure()
+        .executeWith(localityPreferred ? ThreadPools.getWorkerPool() : null)
```

Comment:
this avoids the thread pool when locality is disabled, which is better. But I want to double check if we need to use thread pool here. I saw `Util.blockLocations` eventually calls `getFileStatus` which probably will be a network call to hadoop name node. So thread pool does make sense to me.

Looking at the current caller (`IcebergSplit.getLocations()`) of `Util.blockLocations`  in iceberg-mr module. It doesn't use thread pool. Loop in @rdsr  and @rdblue who might have more context here.

@hililiwei it might be simpler to call this API from `Util` class
```
  public static String[] blockLocations(FileIO io, CombinedScanTask task)
```

This is how Spark code calls it.
```
      if (localityPreferred) {
        Table table = tableBroadcast.value();
        this.preferredLocations = Util.blockLocations(table.io(), task);
      } else {
        this.preferredLocations = HadoopInputFile.NO_LOCATION_PREFERENCE;
      }
```

Reply:
thx @stevenzwu, I mainly refer to the following code: :
https://github.com/apache/iceberg/blob/1a11038baaca9f4cf03674654a6d9a85f39a1ce1/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkMicroBatchStream.java#L135-L144

Reply:
@hililiwei Thanks for the Spark code reference. As I said earlier, using thread pool does make sense to me. I just want to double check the behavior in iceberg-mr module. Hence like to double check with folks with more context.

My 2nd part of the comment is not addressed. please take a look.
```
public static String[] blockLocations(FileIO io, CombinedScanTask task)
```

Reply:
@stevenzwu sorry, I didn't reply 2nd part in time because I'm trying to understand what you mean. But I'm really confused now, the current method I used just is `blockLocations(FileIO io, CombinedScanTask task)`, what should I do?

Reply:
@hililiwei sorry for the confusion. I misread the code. so ignore the 2nd part of the comment.